### DISCUSSION TITLE
'x21'

### DISCUSSION THREAD

**Discussion by digital-prepress** [coloria-dev/coloria#194] [General]
C:\Users\Administrator\PycharmProjects\pythonProject\venv\Scripts\python.exe C:/Users/Administrator/PycharmProjects/pythonProject/k6.py
Traceback (most recent call last):
  File "C:/Users/Administrator/PycharmProjects/pythonProject/k6.py", line 1, in <module>
    import colorio
  File "C:\Users\Administrator\PycharmProjects\pythonProject\venv\lib\site-packages\colorio\__init__.py", line 1, in <module>
    from x21 import __dex__
ModuleNotFoundError: No module named 'x21'

**Comment by nschloe** [comment]
Does
```
pip install x21
```
do anything for you?

**Reply by digital-prepress** [reply]
1)on windows7 ,the x21 package has installed,but wrong to run you demo script
import colorio
import matplotlib.pyplot as plt

illu = colorio.illuminants.d65()
plt.plot(illu.lmbda_nm, illu.data)
plt.xlabel("wavelength [nm]")
plt.show()

--------------------------------------
Traceback (most recent call last):
  File "C:/Users/Administrator/PycharmProjects/pythonProject/q2.py", line 1, in <module>
    import colorio
  File "C:\Users\Administrator\PycharmProjects\pythonProject\venv\lib\site-packages\colorio\__init__.py", line 1, in <module>
    from x21 import __dex__
ModuleNotFoundError: No module named 'x21'



2)on ubuntu2004lts,everything is ok

**Reply by nschloe** [reply]
Does
```
python3 -c "import x21"
```
give the same error for you?

**Reply by digital-prepress** [reply]
yes, running python3.8 32bit  on windows7  ,all the packages installed with pycharm 2021.3.3 community edtion

Python 3.8.0 (tags/v3.8.0:fa919fd, Oct 14 2019, 19:21:23) [MSC v.1916 32 bit (Intel)] on win32

**Reply by nschloe** [reply]
Ah, thanks for the concise info. I think I know what the problem is. Let me try and fix it.

**Reply by nschloe** [reply]
Could you try colorio 0.12.7?

**Reply by digital-prepress** [reply]
$ pip install -v colorio==0.12.7
Using pip 22.0.4 from C:\Users\Administrator\PycharmProjects\pythonProject\venv\lib\site-packages\pip (python 3.8)
Collecting colorio==0.12.7
  Downloading colorio-0.12.7-py3-none-any.whl (1.2 MB)
     ---------------------------------------- 1.2/1.2 MB 2.6 MB/s eta 0:00:00

    Found existing installation: colorio 0.12.2
    Uninstalling colorio-0.12.2:
      Removing file or directory c:\users\administrator\pycharmprojects\pythonproject\venv\lib\site-packages\colorio-0.12.2.dist-info\
      Removing file or directory c:\users\administrator\pycharmprojects\pythonproject\venv\lib\site-packages\colorio\
      Successfully uninstalled colorio-0.12.2
Successfully installed colorio-0.12.7 x21-0.2.4




it works!
